src/pool/srv_iv.c

```diff
@@ -39,35 +39,29 @@ pool_iv_map_ent_size(int nr)
 }
 
 static uint32_t
-pool_iv_prop_ent_size(int nr)
+pool_iv_prop_ent_size(int nr, int nr_ranks)
 {
-	return offsetof(struct pool_iv_entry, piv_prop.pip_acl.dal_ace[nr]);
+	uint32_t total_size;
+
+	total_size = sizeof(struct pool_iv_entry) +
+		     roundup(roundup(nr, 4), 8) +
+		     roundup(nr_ranks * sizeof(int), 8);
```

Comment:
did not get why calculate the size like that, maybe clearer to be sth like:

uint32_t acl_size, svc_size;
struct daos_acl acl;
acl_size = roundup(offsetof(struct daos_acl, acl.dal_ace[nr_aces]), 8);
svc_size = roundup(nr_ranks * sizeof(d_rank_t), 8);
return sizeof(struct pool_iv_entry) + acl_size + svc_size;

Reply:
right, I changed daos_acl to a pointer in pool_iv_entry, but forget to fix here. thanks.